### PR TITLE
Integrate with Azure infrastructure

### PR DESCRIPTION
**Pull request by ArieShout**
Integrate the plugin with Azure infrastructure, this will offer similar user experience across all other Azure Jenkins plugins.

This includes the following changes:

* Allow user to select Azure Credentials -> Resource Group -> Service Fabric in the configuration page.
   * Deploy with Azure resource configuration:   
      
      image
   * Deploy with direct service fabric management endpoint (provided originally):
      
      image

* Integrate with ApplicationInsights to provide usage statistics data

### CODE REVIEW

**Comment by ArieShout** [comment]
One more thing, can we open the `Issues` page for this repository?

File: src/main/resources/com/microsoft/jenkins/servicefabric/ServiceFabricPublisher/config.jelly [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
+            <f:entry title="Resource Group" field="resourceGroupName">
+                <f:select/>
+            </f:entry>
+
+            <f:entry title="Service Fabric" field="serviceFabricName">
+                <f:select/>
+            </f:entry>
+        </f:radioBlock>
+
+        <f:radioBlock name="configureType" value="fill" title="Fill the Service Fabric Management Endpoint"
```
Comment (rapatchi) [inline src/main/resources/com/microsoft/jenkins/servicefabric/ServiceFabricPublisher/config.jelly]: @suhuruli  Can you look into the names such as "Select Service fabric Cluster" and "Fill Service Fabric Management Endpoint". And suggest changes if required ?

Comment (suhuruli) [reply src/main/resources/com/microsoft/jenkins/servicefabric/ServiceFabricPublisher/config.jelly]: LGTM

**Comment by suhuruli** [comment]
Can we merge this PR?
@rapatchi 
@ArieShout

**Pull request merged**